cpmpy/transformations/flatten_model.py

```diff
@@ -578,18 +587,18 @@ def negated_normal(expr):
         elif expr.name == '->':
             # XXX this might create a top-level and
             return expr.args[0] & negated_normal(expr.args[1])
+        elif expr.name == 'not':
+            return expr.args[0]
         else:
             #raise NotImplementedError("negate_normal {}".format(expr))
             # XXX do raise, better safe then sorry
             return expr == 0 # can't do better than this...
 
-    elif expr.name == 'xor':
-        # stay in xor space
-        # only negate last element
-        from ..expressions.globalconstraints import Xor  # avoid circular import
-        return Xor(expr.args[:-1] + [negated_normal(expr.args[-1])])
-
     else: # circular if I import GlobalConstraint here...
+        if hasattr(expr, "decompose_negation"):
+            # for global constraints where the negation of the decomposition is not equivalent
+            # to the negated global constraint (due to auxiliary variables, i.e. Circuit)
+            return Operator('and', expr.decompose_negation())
```

Comment:
we are not very consistent with the use of our builtin-overwritten `any()` and `Operator('and',...)`.

one benefit of `all()` is that it removes constants from the args, one benefit of Operator is that we dont have to import our all, and that we save the effort of a function call and going over the loop...

we were already not consistent, so not for you to fix in this PR, but to keep in mind that we might want to make it consistent at some point...